zork++/src/lib/utils/template.rs

```diff
@@ -0,0 +1,199 @@
+use crate::config_cli::CppCompiler;
+use log::info;
+use std::io::Write;
+use std::path::PathBuf;
+use std::process::Command;
+use std::str::FromStr;
+use std::{
+    fs::{DirBuilder, File},
+    path::Path,
+};
+
+/// Generates a new C++ standarized empty base project
+///        with a pre-designed structure to organize the
+///        user code in a modern fashion way.
+///        Base design for create the project files and folders:
+///            - ./ifc/<project_name>
+///                - hello_zork.<mod_extension>
+///            - ./src/<project_name>
+///                - hello_zork.cpp
+///            - test
+///            - dependencies
+pub fn create_template_project(
+    name_template: &Option<String>,
+    git: bool,
+    compiler: &Option<CppCompiler>,
+) {
+    // TODO required validate or control optionals
+    info!("Create template project");
+    let name = name_template.as_deref().unwrap_or("calculator");
+    let resolved_compiler = compiler.clone().unwrap_or(CppCompiler::CLANG); // Deref ??? and not clone?
+
+    let root_path = Path::new("example");
+    let path_ifc = root_path.join("ifc");
+    let path_src = root_path.join("src");
+    let path_test = root_path.join("test");
+    let path_dependencies = root_path.join("dependencies");
+
+    create_directory(root_path);
+    create_directory(&path_ifc);
+    create_directory(&path_src);
+    create_directory(&path_test);
+    create_directory(&path_dependencies);
+
+    let extension_ifcs = match resolved_compiler {
+        CppCompiler::CLANG => "cppm",
+        CppCompiler::MSVC => "ixx",
+        CppCompiler::GCC => "ixx",
+    };
+    create_file(
+        &path_ifc,
+        format!("{}.{}", "math", extension_ifcs).as_str(),
+        INTERFACE_MOD_FILE,
+    );
+    create_file(&path_src, "main.cpp", MAIN.as_bytes());
+    create_file(&path_src, "math.cpp", SRC_MOD_FILE.as_bytes());
+    create_file(&path_src, "math2.cpp", SRC_MOD_FILE_2.as_bytes());
+
+    let mut zork_conf = ZORK_CONF
+        .replace("<project_name>", name)
+        .replace("<autog_test>", name)
+        .replace("<autogenerated_executable>", name);
+
+    if cfg!(windows) {
+        zork_conf = zork_conf.replace("libcpp", "stdlib")
+    }
+    create_file(
+        &PathBuf::from_str(root_path.to_str().unwrap())
```

Comment:
Replace raw `.unwrap()` for `.expect(msg)`, or better use `if let` pattern in these situations